daft/dataframe/dataframe.py

```diff
@@ -245,6 +245,25 @@ def __iter__(self) -> Iterator[Dict[str, Any]]:
                     row = {key: value[i] for (key, value) in pydict.items()}
                     yield row
 
+    @DataframePublicAPI
+    def to_arrow_iter(self) -> Iterator["pyarrow.Table"]:
+        """
+        Return an iterator of pyarrow tables for this dataframe.
+        """
+        if self._result is not None:
+            # If the dataframe has already finished executing,
+            # use the precomputed results.
+            yield self.to_arrow()
+
+        else:
+            # Execute the dataframe in a streaming fashion.
+            context = get_context()
+            partitions_iter = context.runner().run_iter_tables(self._builder, results_buffer_size=1)
```

Comment:
Also we should make `results_buffer_size` configurable I think. I'm also going to open a PR to have this default to `None` instead